x-pack/test/functional/apps/dataset_quality/dataset_quality_privileges.ts

```diff
@@ -0,0 +1,237 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { DatasetQualityFtrProviderContext } from './config';
+import { getInitialTestLogs, getLogsForDataset } from './data';
+
+export default function ({ getService, getPageObjects }: DatasetQualityFtrProviderContext) {
+  const PageObjects = getPageObjects([
+    'common',
+    'security',
+    'navigationalSearch',
+    'observabilityLogsExplorer',
+    'datasetQuality',
+  ]);
+  const security = getService('security');
+  const synthtrace = getService('logSynthtraceEsClient');
+  const testSubjects = getService('testSubjects');
+  const find = getService('find');
+  const to = new Date(new Date().setDate(new Date().getDate() - 1)).toISOString();
+
+  const apacheAccessDatasetName = 'apache.access';
+  const apacheAccessDatasetHumanName = 'Apache access logs';
+
+  describe('Dataset quality handles user privileges', () => {
+    before(async () => {
+      await PageObjects.observabilityLogsExplorer.setupInitialIntegrations();
+    });
+
+    after(async () => {
+      await PageObjects.observabilityLogsExplorer.removeInstalledPackages();
+    });
+
+    describe('User cannot read any logs-*', () => {
+      before(async () => {
+        // Index logs for synth-* and apache.access datasets
+        await synthtrace.index(getInitialTestLogs({ to, count: 4 }));
+
+        await createDatasetQualityUserWithRole(security, 'dataset_quality_no_read', []);
+        await PageObjects.security.forceLogout();
+        await PageObjects.security.login(
+          'dataset_quality_no_read',
+          'dataset_quality_no_read-password',
+          {
+            expectSpaceSelector: false,
+          }
+        );
+
+        await PageObjects.datasetQuality.navigateTo();
+      });
+
+      after(async () => {
+        await synthtrace.clean();
+        await deleteDatasetQualityUserWithRole(security, 'dataset_quality_no_read');
+      });
+
+      it('shows empty state as user cannot read any dataset', async () => {
+        // Existence of `datasetQualityNoPrivilegesEmptyState` cannot be asserted as without the logs-*-* read privilege,
+        // only the management home screen is shown
+        await testSubjects.existOrFail(
+          PageObjects.datasetQuality.testSubjectSelectors.managementHome
+        );
+      });
+    });
+
+    describe('User can read logs-*', () => {
+      before(async () => {
+        await createDatasetQualityUserWithRole(security, 'dataset_quality_limited_user', [
+          { names: ['logs-*'], privileges: ['read', 'view_index_metadata'] },
+          { names: ['logs-synth*'], privileges: ['read'] }, // No monitor privilege
+          { names: ['logs-apache*'], privileges: ['monitor'] },
+        ]);
+        await PageObjects.security.forceLogout();
```

Comment:
same here

Reply:
Logout here is needed to re-login with a new user.